### DISCUSSION TITLE
[LOGGING] New logging system

### DISCUSSION THREAD

**Discussion by MrThatKid** [TeamRizu/OutFox#286] [03 - Feature Requests]
So, since a new logging system is planned to replace RageLog (The files you see in the Logs folder), we need to figure out what should get logged and where it should be placed.

For example (these aren't set in stone or exhaustive, it's just what I could think of at the moment):

SongLog
- song loading & error messages
- cache-related messages

ThemeLog
- metrics & string errors/warnings
- theme-related lua errors?

EffectLog
- Attack debug
- simfile-related lua errors?

EngineLog
- debug stats
- hardware detected
- debug info from the renderer

**Comment by lomdr** [comment]
The attack log/debug should also log attacks launched by attack mines and any errors related to them (normal attacks, random attacks, and mine attacks)

**Comment by moruzerinho6** [comment]
Would maybe like a TranslationLog

° Shows what line the theme is trying to grab and which theme is trying to and if its trying to get from the theme languages or fallback
° Warns when it fallsback to en.ini if it couldn't find a line from the selected language

**Comment by 48productions** [comment]
Looks like a nice list so far, I like the idea of splitting up the logs to make things a bit easier to find. Only additions I can think of for now:

 - I like the idea of logging simfile lua errors, but I think the name EffectLog doesn't lend itself to simfile errors very well (though tbf I think it's not a bad name for attack debugging). Maybe rename EffectLog to something like SongModsLog, or move simfile lua errors to SongLog?

 - It'd be nice if hardware disconnects/errors showed up in EngineLog as well (say if your pads unplug themselves mid-game, if a configured device (like a joystick or lights driver) can't be opened, etc). Detailed logging info for USB memory cards would be super useful here as well (inserts/removals, errors, etc).

**Comment by fluffy-critter** [comment]
I'd like a log that contains:

* song played
* score / percentage (or whether the song was quit/abandoned early)
* user profile
* step judgements (just the overall counts)

which would be handy for doing stats/progress graphs/etc. on
